core/data.go

```diff
@@ -165,6 +169,67 @@ func (b Bundle) Size() uint64 {
 	return size
 }
 
+func (b Bundle) Serialize() ([]byte, error) {
+	if len(b) == 0 {
+		return []byte{}, nil
+	}
+	if len(b[0].Coeffs) == 0 {
+		return nil, errors.New("invalid bundle: the coeffs length is zero")
+	}
+	size := 0
+	for _, f := range b {
+		if len(f.Coeffs) != len(b[0].Coeffs) {
+			return nil, errors.New("invalid bundle: all chunks should have the same length")
+		}
+		size += bn254.SizeOfG1AffineCompressed + encoding.BYTES_PER_SYMBOL*len(f.Coeffs)
+	}
+	result := make([]byte, size+8)
+	buf := result
+	metadata := uint64(GnarkBundleEncodingFormat) | (uint64(len(b[0].Coeffs)) << 8)
```

Comment:
just informational. The first 64 bytes belong to the header, the least significant byte has the length, the next 63 bytes are for length? I think the maximal chunk length we have is 2^28.

Reply:
Yes, let me add some documentation to make it clear.

The first 8 bytes are header (not 64).
Good callout about the max chunk length, then just this 8 bytes will provide a lot of space for encoding information. Right now it'll just be: 1) what's the encoding format is this? 2) what's the chunk len?